myhoard/web_server.py

```diff
@@ -201,6 +239,8 @@ def _add_routes(self):
             [
                 web.get("/backup", self.backup_list),
                 web.post("/backup", self.backup_create),
+                web.post("/backup_cancel_preserve", self.backup_cancel_preserve),
+                web.post("/backup_preserve", self.backup_preserve),
```

Comment:
Just following the same format as pghoard, rather have a common pattern when actually implementing the preservation.

Anyways, most of the endpoint definitions we have is messy and misleading. For example, restoring a backup is `/PUT status` and sending the stream_id on the body... when ideally it should be `/backup/{stream_id}/restore`. So, its a bit arbitrary to decide which is the best way to expose a functionality through the API , 
refactoring this is not my main focus here :woman_shrugging:  (Even `GET /backup`, since it is not a plural, so wouldn't expect a list of backups )

I can change the method to PUT  (considered it initially but I overthinking since the endpoint only creates a new file, so it can also be POST, but anyways). 

We can focus on refactoring myhoard in other issues.

Reply:
Wow you're right, it is indeed messy. I didn't dig too deep into the other ones. But true, it may be worth considering having a ticket to have a `/v2/...` (keep backward compatibility), start deprecating old endpoints, and some time in the future (major myhoard version), remove them.

Agreed, see what you can do with the new ones to make them "not too messy at a low cost" and we'll see about some major thing in the future for this.

Reply:
I think the API suggested by @Mulugruntz is better.

`/status` is different because that endpoint updates the global status of what the application is doing. You cannot be restoring multiple different backups at the same time and having an API that says you want to update status to restoring backup and give the backup as parameter is imo clearer than having the restoration as a property of a backup. The restoration may also automatically switch to restoring a different backup if the requested one was broken.

For backup preservation this is clearly a property of an individual backup and any number of the backups may be preserved.

Reply:
Or rather than having the separate `/preserve` endpoint I think we should just have `PUT /backup/<stream_id>` operation where the body can contain `{"preserve_until": "<timestamp>"}` value.

Reply:
Did you consider the `PUT /backup/<stream_id>` (or perhaps more appropriately `PATCH /backup/<stream_id>`)? I don't see any obvious downside with that and it would be much more restful approach.